src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -544,24 +583,27 @@ object IEnumerator.Current
 
             public bool MoveNext()
             {
-                if (_version == _builder._version)
+                ReadOnlyCollectionBuilder<T> localBuilder = _builder;
```

Comment:
I understand that's the intent, but that issue did not reach a conclusion yet about actually taking such breaking changes (and I personally don't think we should in these cases where there's so little value in doing so), so it's premature to submit / review a PR that makes those changes.